### PR TITLE
Add a mint amount limit and ownership transfer function to Faucet 

### PR DESCRIPTION
**Pull request by foodaka**

### CODE REVIEW

File: test/faucet.spec.ts [hunk 1]
```diff
@@ -40,6 +42,37 @@ makeSuite('Faucet', (testEnv: TestEnv) => {
     it('Getter isPermissioned should return false', async () => {
       await expect(await faucetOwnable.isPermissioned()).is.equal(false);
     });
+
+    it('Mint function should mint tokens within limit', async () => {
+      const withinLimitThreshold = parseEther('100');
```
Comment (miguelmtzinf) [inline test/faucet.spec.ts:47]: better to fetch it from contract

File: contracts/mocks/testnet-helpers/Faucet.sol [hunk 2]
```diff
@@ -14,10 +15,12 @@ contract Faucet is IFaucet, Ownable {
   // If disabled, anyone can call mint at the faucet, for PoC environments
   bool internal _permissioned;
 
+  // Maximum amount of tokens per mint allowed
+  uint256 private constant MAX_MINT_AMOUNT = 10000;
```
Comment (miguelmtzinf) [inline contracts/mocks/testnet-helpers/Faucet.sol:19]: we can make it public

File: contracts/mocks/testnet-helpers/Faucet.sol [hunk 3]
```diff
@@ -4,6 +4,7 @@ pragma solidity ^0.8.0;
 import {IFaucet} from './IFaucet.sol';
 import {TestnetERC20} from './TestnetERC20.sol';
 import {Ownable} from '@aave/core-v3/contracts/dependencies/openzeppelin/contracts/Ownable.sol';
+import 'hardhat/console.sol';
```
[the remark this answers is not in the document]
Comment (foodaka) [reply contracts/mocks/testnet-helpers/Faucet.sol]: `5166eb05400757d705f83ffb61295fc2c85553d8`

File: contracts/mocks/testnet-helpers/Faucet.sol (hunk 2 above)
Comment (foodaka) [reply contracts/mocks/testnet-helpers/Faucet.sol]: `5166eb05400757d705f83ffb61295fc2c85553d8`

File: test/faucet.spec.ts [hunk 4]
```diff
@@ -40,6 +42,37 @@ makeSuite('Faucet', (testEnv: TestEnv) => {
...
+        deployer,
+      } = testEnv;
+
+      await faucetOwnable
+        .connect(deployer.signer)
+        .mint(dai.address, user.address, withinLimitThreshold);
+      await expect(await dai.balanceOf(user.address)).eq(withinLimitThreshold);
+    });
+
+    it('Mint function should revert with values over 10,000', async () => {
```
[the remark this answers is not in the document]
Comment (foodaka) [reply test/faucet.spec.ts]: `5166eb05400757d705f83ffb61295fc2c85553d8`

File: test/faucet.spec.ts (hunk 1 above)
Comment (foodaka) [reply test/faucet.spec.ts]: `5166eb05400757d705f83ffb61295fc2c85553d8`

**Review by miguelmtzinf — APPROVED** [review]

File: contracts/mocks/testnet-helpers/Faucet.sol [hunk 5]
```diff
@@ -50,4 +57,14 @@ contract Faucet is IFaucet, Ownable {
   function isPermissioned() external view override returns (bool) {
     return _permissioned;
   }
+
+  /// @inheritdoc IFaucet
+  function transferOwnershipOfChild(
```
Comment (foodaka) [inline contracts/mocks/testnet-helpers/Faucet.sol:62]: Nice addition

**Pull request merged**